ark/utils/plot_utils.py

```diff
@@ -6,13 +6,55 @@
 import skimage.io as io
 import matplotlib as mpl
 import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
 
 from skimage.segmentation import find_boundaries
 from skimage.exposure import rescale_intensity
 
 
 # plotting functions
 
+def plot_clustering_result(labeled_image, labels_dict, title,
+                           default_val=None, cmap='tab20', figsize=(10, 10)):
+    """Takes a labeled image and a dictionary that maps cell (labels) to clusters
+    and display an image showing which cells were assigned to which cluster.
+
+    Args:
+        labeled_image (numpy.ndarray):
+            2D numpy array of labeled cell objects.
+        labels_dict (dict):
+            a mapping between labeled cells and their clusters.
+        title (str):
+            The title of the image that will be displayed.
+        default_val (int):
+            Cells that were not assigned to any cluster (according to labels_dict
+            will be translated to this value. This value should not appear in
+            labels_dict.
+        cmap (str):
+            Cmap to use for the image that will be displayed.
+        figsize (tuple):
+            Size of the image that will be displayed.
+    """
+    if not default_val:
+        default_val = max(labels_dict.values()) + 1
+    if default_val in labels_dict.values():
+        raise ValueError('Invalid value for default_val. '
+                         'default_val should not appear as a value in labels_dict.')
+    # copy the given image so that it will not be overwritten
+    img = np.copy(labeled_image)
```

Comment:
This works, but a double nested for loop is going to be slow, especially if people are doing this across all of the images in their cohort. Check out how we used regionprops [here](https://github.com/angelolab/ark-analysis/blob/91610fa6a260d9b315eb273ebace2dc7fe561a94/ark/segmentation/marker_quantification.py#L87) to get all of the coordinates associated with each cell at once

Reply:
I'm not quite sure actually how to make it more efficient. It is a nested for loop, but it goes through each pixel exactly once (n^2 iterations, but actually it is linear in the size of the image). 
If we go in a loop over all cell labels, and then for each label use np indexing to find all entries of `img` equal to that label and change the value to the corresponding cluster label, it shouldn't be faster.

Reply:
The skimage.regionprops function does a lot of optimization under the hood to speed things up. Using the indices produced by regionprops ends up being a lot faster than looping over pixels. I have to admit I'm not actually sure why this is the case; maybe or @alex-l-kong or @vacuousplanet have a more satisfying explanation.

Reply:
@omerbt looking at the source code for `skimage.measure.regionprops` and `find_objects` in `scipy` (which `regionprops` relies on), it looks like the reason why `regionprops` runs faster is because the meat of `find_objects` is optimized to run in C.

Reply:
I changed the implementation (it is now in `relabel_img_array`)